addons/website_sale/models/product_product.py

```diff
@@ -309,116 +273,206 @@ def _prepare_gmc_price_info(self):
-                    'country': country.code,
-                    'price_threshold': utils.gmc_format_price(
-                        delivery['free_over_threshold'], delivery['currency'],
-                    ),
+            case 'meta':
+                return {
+                    'shipping': [{
+                        'country': country.code,
+                        'region': state.code if state else '',
+                        'service': '',
+                        'price': utils.product_feed_format_price(
+                            delivery['price'], delivery['currency']
+                        ),
+                    }
+                    for country, delivery in list(best_delivery_by_country.items())[:100]
+                    for state in (delivery.get('states') or [None])],
                 }
-                for country, delivery in best_delivery_by_country
-                if 'free_over_threshold' in delivery
-            ][:100],  # Apply the limit after looping to include as many results as possible.
-        }
 
-    def _prepare_gmc_stock_info(self):
-        """ Intended to be overridden in stock """
-        self.ensure_one()
-        return {'availability': 'in_stock'}
+    def _prepare_feed_items(self, format='gmc'):
+        """
+        Shared feed item builder for supported platforms: 'gmc', 'meta'.
+        """
+        records = self.with_context(lang=request.lang.code)
+        base_url, delivery_methods_sudo, all_countries, format_product_link = (
+            records._prepare_feed_setup()
+        )
 
-    def _prepare_gmc_additional_info(self):
-        self.ensure_one()
-        gmc_info = {
-            'product_detail': [
-                (attr.attribute_id.name, attr.name)
-                for attr in self.product_template_attribute_value_ids
-            ],
-            'is_bundle': 'yes' if self.type == 'combo' else 'no',
-            'product_type': [
-                category.replace('/', '>')  # google uses a different format
-                for category in (
-                    # up to 5 categories
-                    self.public_categ_ids.sorted('sequence').mapped('display_name')[:5]
-                )
-            ],
-            'custom_label': [
-                (f'custom_label_{i}', tag_name)
-                for i, tag_name in enumerate(
-                    # supports up to 5 custom labels
-                    self.all_product_tag_ids.sorted('sequence').mapped('name')[:5]
-                )
-            ],
+        def build_item(product):
+
+            if not (product._is_variant_possible() and product._prepare_price_info()):
+                return None  # Skip products with no price info
+            base = {
+                **product._prepare_common_feed_items(format_product_link, base_url),
+                **product._prepare_price_info(format=format),
+                **product._prepare_feed_additional_info(format=format),
+                **product._prepare_feed_shipping_info(
```

Comment:
computing the combination dict is quite expensive.

```suggestion
            if not (product._is_variant_possible() and (price_info := product._prepare_price_info())):
                return None  # Skip products with no price info
            base = {
                **product._prepare_common_feed_items(format_product_link, base_url),
                **price_info,
                **product._prepare_feed_additional_info(format=format),
                **product._prepare_feed_shipping_info(
```